output.c

```diff
@@ -80,8 +80,12 @@ static inline void
 output_layout_add(struct cg_output *output, int32_t x, int32_t y)
 {
 	assert(output->scene_output != NULL);
+	bool exists = wlr_output_layout_get(output->server->output_layout, output->wlr_output);
 	struct wlr_output_layout_output *layout_output =
 		wlr_output_layout_add(output->server->output_layout, output->wlr_output, x, y);
+	if (exists) {
+		return;
+	}
```

Comment:
Alternatively one could just `wlr_output_layout_remove` before calls to `wlr_output_layout_add(_auto)`, avoiding the need to check anything.

Reply:
Yeah, but that would re-create the `wl_output` global which I'd prefer not to do.

Reply:
Hmm, good point, that would be a bit annoying on the wire.

Maybe we should replace that assert in scene_output_layout with a an early return to better match how wlr_output_layout works...

Reply:
This was added in https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4380 FWIW.

Reply:
https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4559

(Note that unless we're backporting that patch to wlroots 0.17, this PR is still necessary.)